packages/playwright-test/src/runner.ts

```diff
@@ -243,6 +243,13 @@ export class Runner {
         return true;
       const name = project.name.toLocaleLowerCase();
       unknownProjects!.delete(name);
```

Comment:
This comment is not about your code, but rather the code that used to be here.

I have to admit that I'm struggling with following the logic here. Maybe the naming of the variables is off, maybe the logic itself is unnecessarily complex. Could you go over the large blocks here and add comments explaining what the code is doing? Ideally we should refactor it to make sense, but adding comments is a good start.